### PR TITLE
libwebsocket: remove duplicate default option

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **libwebsocket/***

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by Croydon** [comment]
So many duplicated options. Probably many of us are using a line-duplicating hotkey when adding many options and sometimes we might forget to edit lines? 😄

**Review by Croydon — APPROVED** [review]

File: recipes/libwebsockets/all/conanfile.py [hunk 1]
```diff
@@ -204,7 +203,7 @@ def requirements(self):
             self.requires("libuv/1.38.0")
```
Comment (SpaceIm) [inline recipes/libwebsockets/all/conanfile.py:203]:
```suggestion
            self.requires("libuv/1.40.0")
```

File: recipes/libwebsockets/all/conanfile.py [hunk 2]
```diff
@@ -204,7 +203,7 @@ def requirements(self):
...
 
         if self.options.with_libevent == "libevent":
-            self.requires("libevent/2.1.11")
+            self.requires("libevent/2.1.12")
         elif self.options.with_libevent == "libev":
             self.requires("libev/4.27")
```
Comment (SpaceIm) [inline recipes/libwebsockets/all/conanfile.py:208]:
```suggestion
            self.requires("libev/4.33")
```

File: recipes/libwebsockets/all/conanfile.py [hunk 3]
```diff
@@ -214,13 +213,13 @@ def requirements(self):
             self.requires("miniz/2.1.0")
 
         if self.options.with_libmount:
-            self.requires("libmount/2.33.1")
+            self.requires("libmount/2.36")
 
         if self.options.with_sqlite3:
             self.requires("sqlite3/3.33.0")
```
Comment (SpaceIm) [inline recipes/libwebsockets/all/conanfile.py:219]:
```suggestion
            self.requires("sqlite3/3.34.0")
```

**Review by SpaceIm — COMMENTED** [review]
wolfssl can also be updated to 4.5.0

File: recipes/libwebsockets/all/conanfile.py [hunk 4]
```diff
@@ -418,7 +417,7 @@ def _configure_cmake(self):
         self._cmake.definitions["LWS_WITH_ALSA"] = False
         self._cmake.definitions["LWS_WITH_GTK"] = False
 
-        self._cmake.definitions["LWS_WITH_SYS_SMD"] = False if self.settings.os == "Windows" else True # New in 4.1.4
+        self._cmake.definitions["LWS_WITH_SYS_SMD"] = self.settings.os != "Windows"  # New in 4.1.4
```
Comment (SpaceIm) [inline recipes/libwebsockets/all/conanfile.py:420]:
```suggestion
        if tools.Version(self.version) >= "4.1.0":
            self._cmake.definitions["LWS_WITH_SYS_SMD"] = self.settings.os != "Windows"
```

Option introduced in 4.1.0 actually:
https://github.com/warmcat/libwebsockets/compare/v4.0.21...v4.1.0#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR117

**Review by mathbunnyru — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Comment by SpaceIm** [comment]
@ericLemanissier you can also bump wolfssl: https://github.com/conan-io/conan-center-index/pull/4235#pullrequestreview-567108284

**Review by mathbunnyru — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**